### PR TITLE
feat: add Wix integration app client code

### PR DESCRIPTION
**Pull request by carmelc**
## Purpose

Add the source code of the Wix integration app - https://app.contentful.com/deeplink?link=apps&id=5oytqBOh7xKU40F5c28zQE 

## Testing steps

Install the app, and test it,

## Breaking Changes

No

## Dependencies and/or References

See the documentation at - https://wix-incubator.github.io/cli-apps/packages/contentful/

## Deployment

No

### CODE REVIEW

File: apps/wix/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 Wix.com
```
Comment (whitelisab) [inline apps/wix/LICENSE:3]: Do you want to update the year to `2024` here?

File: apps/wix/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Wix Integration app
+
+The app can be installed using this [link](https://app.contentful.com/deeplink?link=apps&id=5oytqBOh7xKU40F5c28zQE)
```
Comment (whitelisab) [inline apps/wix/README.md:3]: Can you remove this line? We will point people to the marketplace instead to install the app, and your app will have a new id.

File: apps/wix/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "scripts": {
+    "install-ci": "npm ci",
+    "start": "vite",
+    "lint": "prettier --write .",
+    "build": "tsc && vite build",
+    "test": "vitest run",
+    "test:watch": "vitest",
+    "create-app-definition": "contentful-app-scripts create-app-definition",
+    "upload": "contentful-app-scripts upload --bundle-dir ./dist",
```
Comment (whitelisab) [inline apps/wix/package.json:25]: Just an FYI, we will add in a deploy script with your app definition id for our Contentful Marketplace Apps Org in a subsequent PR.

**Review by whitelisab — COMMENTED** [review]
Thanks for opening this PR! Just some small comments for you to review. Let me know if you have any questions.

**Comment by whitelisab** [comment]
@carmelc Can you also confirm that the only location this app renders in is the App Configuration (see location options here: https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). Thank you!

**Comment by carmelc** [comment]
> @carmelc Can you also confirm that the only location this app renders in is the App Configuration (see location options here: https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). Thank you!

Yes, I can confirm this, the app allows the user to view/add/remove access from Wix sites to their CMS data in order to create dynamic pages,

File: apps/wix/LICENSE (hunk 1 above)
Comment (carmelc) [reply apps/wix/LICENSE:3]: Per the legal guidelines at Wix, it should be 2015, as it represents the year copyright was applied.

File: apps/wix/README.md (hunk 2 above)
Comment (carmelc) [reply apps/wix/README.md]: Done

**Review by whitelisab — APPROVED** [review]
@carmelc thanks for the responses and for making changes; this looks good to me!

**Pull request merged**